sql/policy_api.sql

```diff
@@ -128,4 +128,93 @@ CREATE OR REPLACE FUNCTION timescaledb_experimental.show_policies(
     relation REGCLASS)
 RETURNS SETOF JSONB
 AS '@MODULE_PATHNAME@', 'ts_policies_show'
-LANGUAGE C  VOLATILE;
\ No newline at end of file
+LANGUAGE C  VOLATILE;
+
+CREATE OR REPLACE FUNCTION _timescaledb_functions.set_policy_scheduled(hypertable REGCLASS, policy_type TEXT, scheduled BOOL)
+RETURNS INTEGER
+AS $$
+    WITH affected_policies AS (
+        SELECT @extschema@.alter_job(j.id, scheduled => set_policy_scheduled.scheduled)
+        FROM _timescaledb_config.bgw_job j
+            JOIN _timescaledb_catalog.hypertable h ON h.id = j.hypertable_id
+        WHERE j.proc_schema IN ('_timescaledb_internal', '_timescaledb_functions')
+        AND j.proc_name = set_policy_scheduled.policy_type
+        AND j.id >= 1000
+        AND scheduled <> set_policy_scheduled.scheduled
+        AND format('%I.%I', h.schema_name, h.table_name) = set_policy_scheduled.hypertable::text
+    )
+SELECT count(*) FROM affected_policies;
+$$
+LANGUAGE SQL SET search_path TO pg_catalog, pg_temp;
+
+CREATE OR REPLACE FUNCTION _timescaledb_functions.set_all_policy_scheduled(hypertable REGCLASS, scheduled BOOL)
```

Comment:
Nit: Consistency with the other names?

```suggestion
CREATE OR REPLACE FUNCTION _timescaledb_functions.set_all_policies_scheduled(hypertable REGCLASS, scheduled BOOL)
```

Reply:
This would not work with data tiering policies. We should treat that as a timescale policy.
Also, the external API is disable_all_policies and enable_all_policies - as a user I would expect that all policies including any **user defined policy** would get disabled/enabled.

Reply:
@gayyappan Not sure I follow. Why would a different name of the above function not work with data tiering policies?

Reply:
@mkindahl The proc_schema in the sql  is limited to timescaledb_internal/timescaledb_functions. `Not sure I follow. Why would a different name of the above function not work with data tiering policies? `

Reply:
@gayyappan Ah, so you were referring to the actual code for implementing the function, not the function name. It was a comment on my comment, so it wasn't entirely clear.

Reply:
> This would not work with data tiering policies.
Correct. Should it? Data tiering policies reside in a separate extension. Should helper functions in the TimescaleDB function wrap functions in the OSM extension?

> as a user I would expect that all policies including any user defined policy would get disabled/enabled.

The function takes a hypertable as its argument, so it only disables policies which are attached to a hypertable. Maybe the name should change to `{enable,disable}_all_policies_for_hypertable`. From what I can tell, user-defined policies are never associated with a hypertable, so it wouldn't make sense to enable/disable user-defined policies.

> We probably also need to check user's permissions before the user is allowed to enable/disable policies.

All of these functions delegate to `alter_job` under the hood. Does that not perform the permission check that you mention? If not, which permissions would need to be checked?

Reply:
While this is true from a code perspective, from a user perspective, this is 1 database running timescale defined policies - so it should just work out of the box. ```Correct. Should it? Data tiering policies reside in a separate extension. Should helper functions in the TimescaleDB extension wrap functions in the OSM extension? ```
Either we wrap OSM extension or add hooks so that the functionality can be extended (in the same way that the extension extends Postgres functionality).

Reply:
```
All of these functions delegate to alter_job under the hood. Does that not perform the permission check that you mention? If not, which permissions would need to be checked?
```
That could very well be true. You could modify the test to show that a user/role without permissions on the hypertable cannot alter policies.

Reply:
@JamesGuthrie I'll add a follow up to get this to work with OSM extension as well. For this PR, we should verify that permissions are checked by the new apis and add a test case to confirm that.